src/Symbolica.Extensions.Configuration.FSharp/Binder.fs

```diff
@@ -2,52 +2,150 @@ namespace Symbolica.Extensions.Configuration.FSharp
+/// </summary>
+/// <remarks>
+/// By utilising the Reader monad the caller can avoid having to supply an <see cref="IConfiguration" />
+/// until the end of the computation. Making the binding code less cluttered and more declarative.
+/// </remarks>
 type Binder<'a> = Binder of (IConfiguration -> BindResult<'a>)
 
 module Binder =
-    let ofBind b = Binder(fun _ -> b)
-
-    let run (Binder b) = b
-
-    let eval config (b: Binder<'a>) = (run b) config
-
+    /// <summary>Create a <see cref="Binder" /> from a <see cref="BindResult" />.</summary>
+    /// <remarks>
+    /// Effectively this is creating a <see cref="Binder" /> that doesn't need to read from the <see cref="IConfiguration" />.
+    /// </remarks>
+    /// <param name="bindResult">The <see cref="BindResult" /> to create this <see cref="Binder" /> from.</param>
+    /// <returns>A <see cref="Binder" />.</returns>
+    let ofBindResult bindResult = Binder(fun _ -> bindResult)
+
+    /// <summary>Unwraps the <see cref="Binder" />.</summary>
+    /// <returns>The function which was contained inside in the <see cref="Binder" />.</returns>
+    let run (Binder binder) = binder
+
+    /// <summary>Evaluates the <see cref="Binder" /> for the <paramref name="config" />.</summary>
+    /// <param name="config">The <see cref="IConfiguration" /> to bind to.</param>
+    /// <param name="binder">The <see cref="Binder" /> to be evaluated.</param>
+    /// <returns>A <see cref="BindResult" />.</returns>
+    let eval config (binder: Binder<'a>) = (run binder) config
+
+    /// <summary>
+    /// Applies the <see cref="Binder" /> <paramref name="a" /> to the function <paramref name="f" />.
+    /// </summary>
+    /// <remarks>
+    /// This applicative instance accumulates errors.
+    /// If both <paramref name="f" /> and <paramref name="a" /> are <c>Success</c> then the outcome will be <c>Success</c>;
+    /// otherwise the outcome will be a <c>Failure</c> containing all errors.
+    /// </remarks>
+    /// <param name="f">The function to which the value will be applied.</param>
+    /// <param name="a">The value to be applied to the function.</param>
+    /// <returns>A <see cref="Binder" /> containing the result of applying the value to the function.</returns>
     let apply (f: Binder<'a -> 'b>) (a: Binder<'a>) : Binder<'b> =
         Binder(fun config -> BindResult.apply (f |> eval config) (a |> eval config))
 
+    /// <summary>Monadic bind for a <see cref="Binder" />.</summary>
+    /// <remarks>
+    /// If <paramref name="m" /> is <c>Success</c> then the <paramref name="f" /> will be evaluated with
+    /// the data contained in <paramref name="m" />; otherwise the original <c>Failure</c> will be maintained.
+    /// </remarks>
+    /// <param name="f">The function to which the value will be bound.</param>
+    /// <param name="m">The <see cref="Binder" /> that contains the input value to <paramref name="f" />.</param>
```

Comment:
Same as previously, could say `the function <paramref name="f" />`, but this one is actually less important.